packages/taiko-client/pkg/rpc/utils.go

```diff
@@ -290,50 +287,45 @@ func BatchGetBlocksProofStatus(
 	if gErr := g.Wait(); gErr != nil {
 		return nil, gErr
 	}
-
+	parents, err := cli.L2.BatchHeadersByNumbers(ctxWithTimeout, parentIDs)
+	if err != nil {
+		return nil, err
+	}
+	for i := range ids {
+		parentHashes[i] = parents[i].Hash()
+	}
 	// Get the transition state from TaikoL1 contract.
 	transitions, err := cli.TaikoL1.GetTransitions(
 		&bind.CallOpts{Context: ctxWithTimeout},
-		blockIDs,
+		uint64BlockIDs,
 		parentHashes,
 	)
 	if err != nil {
 		return nil, err
 	}
-	highestHeader, err := cli.WaitL2Header(ctxWithTimeout, highestBlockID)
+	_, err = cli.WaitL2Header(ctxWithTimeout, highestBlockID)
 	if err != nil {
 		return nil, err
 	}
-	g, gCtx = errgroup.WithContext(ctxWithTimeout)
+	blockHeaders, err := cli.L2.BatchHeadersByNumbers(ctxWithTimeout, blockIDs)
```

Comment:
check `len(blockHeaders)` must equal to `len(ids)`.

Reply:
Done